autoarray/dataset/imaging/dataset.py

```diff
@@ -167,6 +166,9 @@ def __init__(
 
         self.psf = psf
 
+        if psf.mask.shape[0] % 2 == 0 or psf.mask.shape[1] % 2 == 0:
+            raise exc.KernelException("Kernel2D Kernel2D must be odd")

```

Comment:
I have seen this before but never really understood, why do you want the PSF kernel to be odd?

Reply:
For real space convolution, even x even kernels can induce a half pixel shift depending on where you centre it. This meant that things like the lens model centres had a half pixel shift, in the end, I think.

Alternatively, you could sample the "peak" of the even x even PSF in the central 4 pixels, which means you don't fully sample its shape where it matters most.

This may not matter for an FFT? And certainly wont matter once we do PSF over sampling?

Its definitely a hack I am happy to see removed.